### PR TITLE
v2 get request

### PR DESCRIPTION
**Pull request by SteveOye**
All get request has been implemented

### CODE REVIEW

File: lib/flutter_periculum.dart [hunk 1]
```diff
@@ -267,4 +274,218 @@ class FlutterPericulum {
...
+
+  static Future<List<TransactionResponse>>
+      getMobileV2TransactionsByBankAddressAndAccountNumberForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+    required String smsAddress,
+    required String accountKey,
+  }) async {
+    final uri = Uri.parse(
+        '$BASE_URL/mobile/v2/$insightsOverviewKey/banks/$smsAddress/$accountKey/transactions');
```
Comment (danishperic) [inline lib/flutter_periculum.dart]: this should be bankKey instead of smsAddress

File: lib/flutter_periculum.dart [hunk 2]
```diff
@@ -267,4 +274,218 @@ class FlutterPericulum {
...
+      log(e.toString());
+      rethrow;
+    }
+  }
+
+  static Future<List<TransactionResponse>>
+      getMobileV2TransactionsByBankAddressAndAccountNumberForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+    required String smsAddress,
```
Comment (danishperic) [inline lib/flutter_periculum.dart]: bankKey

File: lib/flutter_periculum.dart [hunk 3]
```diff
@@ -267,4 +274,218 @@ class FlutterPericulum {
...
+
+  static Future<InsightsResponse>
+      getMobileV2InsightsByBankAddressAndAccountNumberForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+    required String smsAddress,
+    required String accountKey,
+  }) async {
+    final uri = Uri.parse(
+        '$BASE_URL/mobile/v2/$insightsOverviewKey/banks/$smsAddress/$accountKey/insights');
```
Comment (danishperic) [inline lib/flutter_periculum.dart]: bankKey instead of smsAddress

**Review by danishperic — CHANGES_REQUESTED** [review]

File: lib/flutter_periculum.dart (hunk 1 above)
Comment (SteveOye) [reply lib/flutter_periculum.dart]: Oh okay

File: example/lib/v2.dart [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                  ),
+                  ElevatedButton(
+                    onPressed: () async {
+                      setState(() {
+                        responseOutput = "GetInsightV2 -> loading.. ";
+                      });
+                      var response;
+                      try {
+                        response =
+                            await FlutterPericulum.getMobileV2InsightOverview(
```
Comment (danishperic) [inline example/lib/v2.dart:75]: can you please rename method to "GetMobileInsightsOverviewByOverviewKeyAsync" ?

File: example/lib/v2.dart [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                  ),
+                  ElevatedButton(
+                    onPressed: () async {
+                      setState(() {
+                        responseOutput = "GetBankInsight -> loading.. ";
+                      });
+                      var response;
+                      try {
+                        response = await FlutterPericulum
+                            .getMobileV2BanksForInsightOverviewKey(
```
Comment (danishperic) [inline example/lib/v2.dart:99]: can you please rename method to "GetBanksForMobileInsightsOverviewKeyAsync" ?

File: example/lib/v2.dart [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                  ),
+                  ElevatedButton(
+                    onPressed: () async {
+                      setState(() {
+                        responseOutput = "GetLendersOverview -> loading.. ";
+                      });
+                      var response;
+                      try {
+                        response = await FlutterPericulum
+                            .getMobileV2LendersForInsightOverviewKey(
```
Comment (danishperic) [inline example/lib/v2.dart:123]: can you please rename method to "GetLendersForMobileInsightsOverviewKeyAsync" ?

File: example/lib/v2.dart [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+                  ),
+                  ElevatedButton(
+                    onPressed: () async {
+                      setState(() {
+                        responseOutput = "GetTransactionInsights -> loading.. ";
+                      });
+                      var response;
+                      try {
+                        response = await FlutterPericulum
+                            .getMobileV2TransactionsByBankAddressAndAccountNumberForInsightOverviewKey(
```
Comment (danishperic) [inline example/lib/v2.dart:147]: can you please rename method to "GetSMSTransactionsByBankKeyAndAccountKeyForInsightsOverviewKeyAsync" ?

File: example/lib/v2.dart [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+                  ),
+                  ElevatedButton(
+                    onPressed: () async {
+                      setState(() {
+                        responseOutput = "GetInsights -> loading.. ";
+                      });
+                      var response;
+                      try {
+                        response = await FlutterPericulum
+                            .getMobileV2InsightsByBankAddressAndAccountNumberForInsightOverviewKey(
```
Comment (danishperic) [inline example/lib/v2.dart:173]: can you please rename method to "GetMobileInsightsByBankKeyAndAccountKeyForInsightsOverviewKeyAsync" ?

**Review by danishperic — CHANGES_REQUESTED** [review]

**Pull request merged**